cmake/modules/AddSwiftHostLibrary.cmake

```diff
@@ -17,10 +17,21 @@
 #
 # Remove once rdar://102202478 is fixed.
 function(target_link_swift_syntax_libraries TARGET)
-  target_link_libraries(${TARGET} ${ARGN})
-
   cmake_parse_arguments(ARGS "PUBLIC;PRIVATE;INTERFACE" "" "" ${ARGN})
-  foreach(DEPENDENCY ${ARGS_UNPARSED_ARGUMENTS})
+  set(link_type)
+  if(ARGS_PUBLIC)
+    set(link_type PUBLIC)
+  elseif(ARGS_PRIVATE)
+    set(link_type PRIVATE)
+  elseif(ARGS_INTERFACE)
+    set(link_type INTERFACE)
+  endif()
```

Comment:
Should it be an error if you omit the link type?

Reply:
https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-both-a-target-and-its-dependents
Since it's optional for `target_link_libraries`, I'm leaving it as is for now

Reply:
Fair enough, my main concern would be if a new link type was added that wasn't handled here, it may get silently dropped